`Development`: Add documentation for modeling exercises (#5369)
ls1intum/Artemis: docs/user/exercises/modeling.rst

- *Inclusion in course score calculation*: Option that determines whether or not to include exercise in course score calculation
- *Points*: Total point of an exercise
- *Bonus Points*: Bonus points of an exercise
- *Diagram Type*: Type of diagram that is used throughout an exercise

    .. figure:: modeling/create-modeling-exercise-form-1.png
              :align: center

.. note::
   Fields marked with red are mandatory to be filled.

.. note::
   - Field **Diagram Type** determines the components that students/instructors can use while working on the exercise.
   - For example: If the instructor selects Class diagram as its Diagram Type, users (instructors/students) will now only be able to use components of class diagram throughout the exercise.

    .. figure:: modeling/class-diagram-diagram-type.png
              :align: center

Following screenshot illustrates the second section of the form. It consists of:

- *Enable automatic assessment suggestions*: Option that determines whether or not to enable assessment suggestions automatically
- *Problem Statement*: Field to state problem statement of an exercise

> - *Assessment Instructions*: Instructions for instructor while assessing the submission

Reviewer: Fine for now, should be changed to a link to the dedicated assessment documentation page once it exists.

    .. figure:: modeling/create-modeling-exercise-form-2.png
              :align: center

.. note::
    If you are not clear about any of the fields, you can access to the field hints by hovering on |hint| icon, when available.

    .. figure:: modeling/create-modeling-exercise-form-hint.png
              :align: center

Following screenshot illustrates the last section of the form. It consists of:

- *Example Solution*: Example solution of an exercise
- *Example Solution Explanation*: Explanation of example solution of an exercise
- *Example Solution Publication Date*: Date after which example solution is accessible for students

    .. figure:: modeling/create-modeling-exercise-form-3.png
              :align: center

Once you are done defining the schema of an exercise, you can now create an exercise by clicking on |save| button.
You will then be redirected to **Example Submissions for Assessment Training** Page

    .. figure:: modeling/example-submission-for-assessment-training.png
              :align: center

> In this page, you can either *Create Example Submission* or *Use as Example Submission* for Assessment Training.

Reviewer: ```suggestion
In this page, you can either *Create Example Submission* or *Use as Example Submission* for Assessment Training.
Example submission can be used to assess the submissions of students semi-automatically.
Artemis uses those submissions to automatically apply the known assessment comments to similar model elements in other submissions as well.
```
I suggest moving the explanation what this page is used for to a place before the description of the possible interactions on the page.

              :align: center

Following screenshot illustrates the first section of the form. It consists of:

- *Title*: Title of an exercise
- *Categories*: Category of an exercise
- *Difficulty*: Difficulty of an exercise. (No level, Easy, Medium or Hard)
- *Mode*: Solving mode of an exercise. *This cannot be changed afterwards* (Individual or Team)
- *Release Date*: Date after which students can access the exercise
- *Due Date*: Date till when students can access the exercise
- *Assessment Due Date*: Date till when instructors can assess the exercise submitted by students
- *Inclusion in course score calculation*: Option that determines whether or not to include exercise in course score calculation
- *Points*: Total point of an exercise
- *Bonus Points*: Bonus points of an exercise
- *Diagram Type*: Type of diagram that is used throughout an exercise

    .. figure:: modeling/create-modeling-exercise-form-1.png
              :align: center

.. note::
   Fields marked with red are mandatory to be filled.

.. note::
   - Field **Diagram Type** determines the components that students/instructors can use while working on the exercise.

>    - For example: If the instructor selects Class diagram as its Diagram Type, users (instructors/students) will now only be able to use components of class diagram throughout the exercise.

Reviewer: ```suggestion
   - For example: If the instructor selects class diagram as its diagram type, users (instructors/students) will now only be able to use components of class diagram throughout the exercise.
```

- When the exercise is released students can work on the exercise.

    .. figure:: modeling/modeling-exercise-card-student-view.png
              :align: center

- They can start the exercise by clicking the |start| button.

- Once they start the exercise, they will now have the option to work on it in an online modeling editor by clicking on  the |open-modeling-editor| button.

- The screenshot below depicts the online modeling exercise interface for students. They can read the Problem Statement, work on the online editor and also provide an explanation to their solutions, if needed.

    .. figure:: modeling/modeling-exercise-students-interface.png
              :align: center

Assessment
----------

When the due date is over you can assess the submissions.

- To assess the submissions, first click on Assessment Dashboard

    .. figure:: modeling/assessment-dashboard.png
              :align: center

- Then click on Submissions of the modeling exercise

    .. figure:: modeling/exercise-dashboard.png
              :align: center

- You will then be redirected to *Submissions* Dashboard

    .. figure:: modeling/submissions-dashboard.png
              :align: center

> - Click on |assess-submission| button of specific student. You will then be redirected to the assessment page where you will be able to assess submission of that student.

Reviewer: I don’t think you are redirected to the submissions page, but instead are on an exercise assessment dashboard.

I think it would be enough the replace this part by something like 
‘The assessment of modeling exercises can be reached as usual via the assessment dashboard’
with ‘assessment dashboard’ being a link to the assessment documentation page (or a dummy ToDo link for now as that page not yet exists).

> - You can now start assessing the elements of the model by double clicking it. Once you double click, you will get an assessment modal where you can assign points, feedback and navigate through all other assessable components.

Reviewer: ```suggestion
- You can now start assessing the elements of the model by double clicking it. Once you double click, you will get an assessment dialog where you can assign points, feedback and navigate through all other assessable components.
```
I think being more generic in modal vs. dialogue is better here for the end user documentation.

    .. figure:: modeling/assessment-modal.png
              :align: center

> - Alternatively, you can also assess the diagram by dragging and dropping assessment instructions from *Assessment Instructions* section.

Reviewer: ```suggestion
- Alternatively, you can also assess the diagram by dragging and dropping assessment instructions from the *Assessment Instructions* section.
```

    .. figure:: modeling/assessment-instruction.png
              :align: center

> - Feedback to the entire submission can also be added by clicking on |add-new-feedback| button.

Reviewer: ```suggestion
- Feedback to the entire submission can also be added by clicking on the |add-new-feedback| button.
```

Modeling exercise
=================
.. contents:: Content of this document
    :local:
    :depth: 3

Overview
--------

Conducting a Modeling exercise consists of 3 steps:

1. **Instructor prepares exercise:** Creates and configures the modeling exercise in Artemis.
2. **Student solves exercise:** Student works on the exercise and submits the solution.

> 3. **Instructor assesses submissions:** Reviews the submitted exercises and creates results for the students.

Reviewer 2: ```suggestion
3. **Tutors assesses submissions:** Reviews the submitted exercises and creates results for the students.
```

Setup
-----

The following sections describe the supported features and the process of creating a new modeling exercise.

- Open |course-management|.
- Navigate into **Exercises** of your preferred course.

    .. figure:: modeling/course-management-course-dashboard.png

>               :align: center

Reviewer 2: I'd leave this out, since it applies to all exercises.

Create new modeling exercise
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

- Click on **Create new modeling exercise**

    .. figure:: modeling/create-new-modeling-exercise.png
              :align: center

The following screenshot illustrates the first section of the form. It consists of:

- **Title**: Title of an exercise
- **Categories**: Category of an exercise
- **Difficulty**: Difficulty of an exercise. (No level, Easy, Medium or Hard)
- **Mode**: Solving mode of an exercise. *This cannot be changed afterwards* (Individual or Team)
- **Release Date**: Date after which students can access the exercise

> - **Due Date**: Date till when students can access the exercise

Reviewer 3: Date till when students can work on the exercise.

Students can access the exercise even after the deadlien

> - **Assessment Due Date**: Date till when instructors can assess the exercise submitted by students

Reviewer 3: Technically the instructors can assess the exercise even after the Assessment Due Date.
I'd prefer, `Date after which students can view the feedback of the assessments from the instructors` or something else

- **Inclusion in course score calculation**: Option that determines whether or not to include exercise in course score calculation

> - **Points**: Total point of an exercise

Reviewer 4: ```suggestion
- **Points**: Total points of an exercise
```

- **Bonus Points**: Bonus points of an exercise
- **Diagram Type**: Type of diagram that is used throughout an exercise

    .. figure:: modeling/create-modeling-exercise-form-1.png
              :align: center

.. note::
   Fields marked with red are mandatory to be filled.

.. note::
   - The field **Diagram Type** determines the components that students/instructors can use while working on the exercise.
   - This option cannot be changed after creating the exercise.

>    - For example: If the instructor selects class diagram as its diagram type, users (instructors/students) will now only be able to use components of class diagram throughout the exercise.

Reviewer 4: ```suggestion
   - For example: If the instructor selects class diagram as its diagram type, users (instructors/students) will now only be able to use components of class diagrams throughout the exercise.
```

    .. figure:: modeling/class-diagram-diagram-type.png
              :align: center

The following screenshot illustrates the second section of the form. It consists of:

- **Enable automatic assessment suggestions**: When enabled, Artemis tries to automatically suggest assessments for diagram elements based on previously graded submissions for this exercise.

> - **Problem Statement**: The task description of the exercise as seen by students

Reviewer 3: Missing full stop at the end of the sentence

> - **Assessment Instructions**: Instructions for instructor while assessing the submission

Reviewer 4: ```suggestion
- **Assessment Instructions**: Instructions for instructors while assessing the submission
```

> - **Assessment Instructions**: Instructions for instructors while assessing the submission

Reviewer 3: Missing full stop at the end of the sentence

    .. figure:: modeling/create-modeling-exercise-form-2.png
              :align: center

.. note::
    If you are not clear about any of the fields, you can access additional hints by hovering over the |hint| icon for many of them.

    .. figure:: modeling/create-modeling-exercise-form-hint.png
              :align: center

The following screenshot illustrates the last section of the form. It consists of:

> - **Example Solution**: Example solution of an exercise

Reviewer 3: Missing full stop at the end of the sentence

> - **Example Solution Explanation**: Explanation of the example solution

Reviewer 3: Missing full stop at the end of the sentence

> - **Example Solution Publication Date**: Date after which the example solution is accessible for students. Leave empty to not publish the solution.

Reviewer 3: Leave empty not to publish the solution

> - **Example Solution Publication Date**: Date after which the example solution is accessible for students. If you leave this field empty, the solution will not be published.

Reviewer 2: ```suggestion
- **Example Solution Publication Date**: Date after which the example solution is accessible for students. If you leave this field empty, the solution will only. be published to tutors.
```

    .. figure:: modeling/create-modeling-exercise-form-3.png
              :align: center

Once you are done defining the schema of an exercise, you can now create an exercise by clicking on |save| button.
You will then be redirected to **Example Submissions for Assessment Training** Page

    .. figure:: modeling/example-submission-for-assessment-training.png
              :align: center

In this page, you can either *Create Example Submission* or *Use as Example Submission* for Assessment Training.

> Example submission can be used to assess the submissions of students semi-automatically.

Reviewer 4: ```suggestion
Example submissions can be used to assess the submissions of students semi-automatically.
```

Artemis uses those submissions to automatically apply the known assessment comments to similar model elements in other submissions as well.

- Select |create-example-submission| if you want to create an example submission from scratch.
- Alternatively, after the exercise already started, you can also use some submissions submitted by students as an example submission. For that, click on |use-as-example-submission|

.. note::
    Artemis uses semi-automatic grading of modeling exercises using machine learning.
    You can hence train the model by selecting *Use in Assessment Training* checkbox while creating an example submission.

    .. figure:: modeling/use-in-assessment-training.png
              :align: center

Import Modeling Exercise
^^^^^^^^^^^^^^^^^^^^^^^^

> - Alternatively, you can also import modeling exercise from the existing one by clicking on **Import Modeling Exercise**

Reviewer 3: Missing full stop at the end of the sentence

    .. figure:: modeling/import-modeling-exercise.png
              :align: center

- An import modal will prompt up, where you will have an option to select and import previous modeling exercises from the list by clicking on |import| button.

    .. figure:: modeling/import-modeling-exercise-modal.png
              :align: center

> - Once you import one of the exercise, you will then be redirected to form which is similar to *Create new modeling exercise* form with all the fields filled from imported exercise. You can now modify the fields as per your necessity to create a new Modeling Exercise.

Reviewer 3: redirected to a form

Result
^^^^^^

    .. figure:: modeling/course-dashboard-exercise-modeling.png
              :align: center

- Click the |edit| button of the modeling exercise and adapt the interactive problem statement. There you can also set release and due dates.
- Click the |scores| button to see the scores achieved by the students.
- Click the |participation| button to see the list of students participated in the exercise.
- Click the |submission| button to see the list of submission submitted by students.
- Click the |example-submission| button to modify/add example submission of the exercise.
- Click the |delete| button to delete the exercise.

> - You can get an overview of the exercise by clicking on the title.

Reviewer 2: This is also more or less the same for all exercises

> Student Submission

Reviewer 2: Maybe move this to an own subpage (and the rest into an other?)
This way, a students don't have to read through half the page before finding the relevant information for them

Modeling exercise
=================
.. contents:: Content of this document
    :local:
    :depth: 2

Overview
--------

Conducting a Modeling exercise consists of 3 steps:

1. **Instructor prepares exercise:** Creates and configures the modeling exercise in Artemis.
2. **Student solves exercise:** Student works on the exercise and submits the solution

> 3. **Instructor assesses submissions:** Review the submitted exercises and create results for the students.

Reviewer 5: ```suggestion
3. **Instructor assesses submissions:** Reviews the submitted exercises and creates results for the students.
```
Coherent with the 3rd person from the 1st point.

Setup
--------

The following sections describe the supported features and the process of creating a new modeling exercise.

- Open |course-management|
- Navigate into **Exercises** of your preferred course

    .. figure:: modeling/course-management-course-dashboard.png
              :align: center

- Click on **Create new modeling exercise**

    .. figure:: modeling/create-new-modeling-exercise.png
              :align: center

  Result: **Modeling Exercise**

    .. figure:: modeling/course-dashboard-exercise-modeling.png
              :align: center

- Click the |edit| button of the modeling exercise and adapt the interactive problem statement. There you can also set release and due dates.
- You can get an overview of the exercise by clicking on the title.

Student Submission
------------------

- When the exercise is released students can work on the exercise.

    .. figure:: modeling/modeling-exercise-card-student-view.png
              :align: center

- They can start the exercise by clicking the |start| button.

- Once they start the exercise, they will now have option to work on it in online modeling editor by clicking on  the |open-modeling-editor| button.

> - The screenshot below depicts the online modeling exercise interface for students. They can read the Problem Statement, work on the online editor and also provide explanation to their solutions, if needed.

Reviewer 4: ```suggestion
- The screenshot below depicts the online modeling exercise interface for students. They can read the Problem Statement, work on the online editor and also provide an explanation to their solutions, if needed.
```
